expression/builtin_regexp_util.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package expression
+
+import (
+	"regexp"
+	"unicode/utf8"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tidb/util/chunk"
+)
+
+// Parameters may be const or ignored by the user, so different situations should be considered
+// We can handle parameters more easily with this struct.
+//
+// isNull field shows if user ignores this param in sql
+// for example:
+//
+//	select regexp_like("123", "123", "m"), here isNull field is false for the third parameter
+//	select regexp_like("123", "123"), here isNull field is true for the third parameter
+type regexpParam struct {
+	constStrVal   string
+	constIntVal   int64
+	defaultIntVal int64 // default value when isNull is true
+	isConst       bool
+	isNull        bool
+	col           *chunk.Column
+}
+
+func (re *regexpParam) getCol() *chunk.Column {
```

Comment:
It seems you should not expose `re.col`, since `re.col` itself maybe invalid

Reply:
> It seems you should not expose `re.col`, since `re.col` itself maybe invalid

re.col will always be valid, because the function will raise error and not continue to execute when we fail to create re.col

Reply:
> It seems you should not expose `re.col`, since `re.col` itself maybe invalid

I will delete this function as it is not used.